doc/contributing.rst

```diff
@@ -158,5 +158,81 @@ Happy testing!
 Connecting new machine learning libraries
 =========================================
 
-Coming soon - please stay tuned!
-
+Hosting the library
+~~~~~~~~~~~~~~~~~~~
+
+Each extension created should be a stand-alone repository, compatible with the
+`OpenML repository <https://github.com/openml/openml-python>`_.
+The extension repository should work off-the-shelf with openml-python installed.
+
+Create a `public Github repo <https://help.github.com/en/articles/create-a-repo>`_ with
+the following directory structure:
+
+::
+
+| [repo name]
+|   |-- extensions
+|   |    |-- [extension name]
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+|   |    |    |-- config.py (optionally)
+|   |    |-- sklearn
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+|   |    |-- keras
+|   |    |    |-- __init__.py
+|   |    |    |-- extension.py
+
+
+
+Content of the Library
+~~~~~~~~~~~~~~~~~~~~~~
+
+To get an understanding of what OpenML does and why support of the latest
+machine learning libraries can be beneficial for the community, please refer
+`here <https://openml.github.io/OpenML/>`_.
+
+Various `examples <https://openml.github.io/openml-python/develop/examples/index.html>`_
+can show how OpenML works in
+practice.
+Moreover `this example <https://openml.github.io/openml-python/develop/examples/
+flows_and_runs_tutorial.html>`_
+shows how scikit-learn currently works with OpenMl as an extension. The
+`sklearn extension <https://github.com/openml/openml-python/tree/develop/openml/extensions>`_
+packaged with the `openml-python <https://github.com/openml/openml-python/issues/730>`_
+repository can be used as a template/benchmark to build the extension.
+
+
+API
++++
+* The extension scripts should be importing `openml` and be able to execute any
+  function from the OpenML `API <https://openml.github.io/openml-python/develop/api.html#>`_.
+* The extension class defined here should inherit the `openml extension
+  class <https://openml.github.io/openml-python/develop/generated/openml.extensions.
+  Extension.html#openml.extensions.Extension>`_.
+* The class should have the functions from `class Extension` overloaded as required.
+  The `Sklearn Extension API <https://openml.github.io/openml-python/develop/generated/
+  openml.extensions.sklearn.SklearnExtension.html#openml.extensions.sklearn.SklearnExtension>`_
+  is a good checklist to follow.
+
+
+Interfacing with OpenML
++++++++++++++++++++++++
+Once the new extension class has been defined, the openml module to `register extensions
+<https://openml.github.io/openml-python/develop/generated/openml.extensions.register_
+extension.html#openml.extensions.register_extension>`_ can be called to allow OpenML
+interface the new extension. For example, the `sklearn extension registering
+<https://github.com/openml/openml-python/blob/develop/openml/extensions/sklearn/
+extension.py#L1744>`_.
```

Comment:
Could you please not refer to specific lines? They might change in the future and make the docs wrong. Could you instead please try to move the registring of the extension to `__init__.zy` and simply link that file?

Reply:
Are you referring to [this](https://github.com/openml/openml-python/blob/develop/openml/extensions/sklearn/__init__.py) file?
So in this file, I'll import `register_extension` and call `register_extension(SklearnExtension)`?
And remove `register_extension(SklearnExtension)` from the current file it is in?

Reply:
Yes, thank you. I'm actually only seeing this now. You're using the full URL, but you can simply specify to which part of the documentation a link should go, see for example here: https://github.com/openml/openml-python/blob/develop/doc/index.rst

Then the exact links are auto-generated and automatically point to the right branch (develop or master).

Reply:
Hi,
I guess I understand what you mean here. However, in this section [here](https://github.com/openml/openml-python/blob/develop/doc/index.rst#example) where the examples gallery is linked, the generated URL links to [this](https://github.com/openml/openml-python/blob/develop/doc/examples/index.html). 
Whereas, what we would want is probably [this](https://openml.github.io/openml-python/develop/examples/index.html). 
Is that so?